sw/host/opentitanlib/src/transport/hyperdebug/mod.rs

```diff
@@ -331,6 +344,41 @@ impl<T: Flavor> Hyperdebug<T> {
             )),
         }
     }
+
+    fn get_cmsis_google_capabilities(&self) -> Result<u16> {
+        let Some(cmsis_interface) = self.cmsis_interface else {
+            // Since this debugger does not advertise any CMSIS USB interface at all, report no
+            // Google CMSIS extension capabilites.
+            return Ok(0);
+        };
+        if let Some(capabilities) = self.cmsis_google_capabilities.get() {
+            // Return cached value.
+            return Ok(capabilities);
+        }
+        let cmd = [
+            Self::CMSIS_DAP_CUSTOM_COMMAND_GOOGLE_INFO,
+            Self::GOOGLE_INFO_CAPABILITIES,
+        ];
+        self.inner
+            .usb_device
+            .borrow()
+            .write_bulk(cmsis_interface.out_endpoint, &cmd)?;
+        let mut resp = [0u8; 64];
+        let bytecount = self
+            .inner
+            .usb_device
+            .borrow()
+            .read_bulk(cmsis_interface.in_endpoint, &mut resp)?;
+        let resp = &resp[..bytecount];
+        ensure!(
+            bytecount >= 4 && resp[0] == Self::CMSIS_DAP_CUSTOM_COMMAND_GOOGLE_INFO && resp[1] >= 2,
```

Comment:
maybe document what the `resp[1] >= 2` is? I assume that this is the number of bytes that follow the header?

Reply:
Done